sdcm/nemesis.py

```diff
@@ -1831,14 +1831,14 @@ def repair_nodetool_rebuild(self):
 
     def disrupt_nodetool_cleanup(self):
         # This fix important when just user profile is run in the test and "keyspace1" doesn't exist.
-        test_keyspaces = self.cluster.get_test_keyspaces()
+        # test_keyspaces = self.cluster.get_test_keyspaces()
 
         # Inner disrupt function for ParallelObject
         def _dis_nodetool_cleanup(node):
             InfoEvent('NodetoolCleanupMonkey %s' % node).publish()
             with adaptive_timeout(Operations.CLEANUP, node, timeout=HOUR_IN_SEC * 48):
-                for keyspace in test_keyspaces:
-                    node.run_nodetool(sub_cmd="cleanup", args=keyspace)
+                # for keyspace in test_keyspaces:
+                node.run_nodetool(sub_cmd="cleanup")
```

Comment:
So where is the parallelism here?
It's supposed to be "for node in db_nodes:"

Reply:
The parallelism is done by parallelObject.

threads = ParallelObject(self.cluster.nodes, num_workers=min(32,
len(self.cluster.nodes)))
thread_results = threads.run(_dis_nodetool_cleanup,
ignore_exceptions=True, unpack_objects=True)

Reply:
Somehow the lines below didn't show up when I looked at it yesterday (or I saw only the last commit)

Reply:
Does it work?
Do we see the cleanup running in parallel on all nodes?

Reply:
Yes It worked, but for some reason it also tried to run cleanup several times on the same node.
This was unexpected behaviour.
Following the code it should run in parallel on all nodes.
Last commit removed looping over keyspaces, so now we should have one cleanup command per node.
Testing in docker seems to fail for other reason.
I'm launching another test run in aws.

Reply:
we see all the 3 nodes having a call to start `nodetool cleanup` in parallel.

the problem I see is that, we are not waiting for the cleanup to end, before "ending the nemesis" and we eventually started the next nemesis before the previous cleanup ended, causing an exception